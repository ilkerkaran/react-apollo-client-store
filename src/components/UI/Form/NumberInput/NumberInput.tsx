import React from 'react';

const NumberInput = ({
  className, inputName,
  label,
  defaultValue = undefined,
  isChanged = false,
  onChange
}: {className:string,
  inputName:string,
  defaultValue:string | undefined
  label:string,
  isChanged:boolean,
  onChange:(e:any) => any}) => (
    <div className={className}>
      <input
        type="number"
        className={isChanged ? 'Input Changed' : 'Input'}
        defaultValue={defaultValue}
        placeholder={label}
        onChange={onChange}
        name={inputName}
      />
    </div>
);

export default NumberInput;
