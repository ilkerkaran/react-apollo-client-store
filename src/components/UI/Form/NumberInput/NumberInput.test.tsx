import React from 'react';
import renderer from 'react-test-renderer';

import NumberInput from './NumberInput';

describe('<NumberInput /> snapshot ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<NumberInput
        className=""
        defaultValue=""
        inputName=""
        isChanged={false}
        label="label"
        onChange={() => {}}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
