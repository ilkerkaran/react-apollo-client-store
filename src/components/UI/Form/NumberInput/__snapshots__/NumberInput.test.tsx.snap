// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput /> snapshot  renders correctly 1`] = `
<div
  className=""
>
  <input
    className="Input"
    defaultValue=""
    name=""
    onChange={[Function]}
    placeholder="label"
    type="number"
  />
</div>
`;
